packages/service-provider-core/src/service-provider.ts

```diff
@@ -2,8 +2,21 @@ import Readable from './readable';
 import Writable from './writable';
 import Closable from './closable';
 import Admin from './admin';
+import makePrintableBson from './printable-bson';
+import bson from 'bson';
 
 /**
  * Interface for all service providers.
  */
 export default interface ServiceProvider extends Readable, Writable, Closable, Admin {}
+
+export class ServiceProviderCore {
+  public bsonLibrary;
+  constructor(bsonLibrary) {
+    if (bsonLibrary === undefined) {
```

Comment:
We could chose to throw an `InternalError` here instead - what do you prefer @lrlna @mcasimir ?

Reply:
InternalError sounds great to me!